kyo-stm/shared/src/main/scala/kyo/TTable.scala

```diff
@@ -0,0 +1,280 @@
+  */
+abstract class TTable[Fields]:
+
+    /** The type of record IDs for this table. Represented as an opaque Int subtype to provide type safety. */
+    type Id <: Int
+
+    object Id:
+        given Flat[Id] = Flat.derive
+
+    /** Converts a raw Int to the table's ID type. Should only be used internally by the implementation.
+      *
+      * @param id
+      *   The raw integer ID value
+      * @return
+      *   The typed ID value
+      */
+    def unsafeId(id: Int): Id
+
+    /** Retrieves a record by its ID.
+      *
+      * @param id
+      *   The ID of the record to retrieve
+      * @return
+      *   The record if found, None otherwise, within the STM effect
+      */
+    def get(id: Id)(using Frame): Maybe[Record[Fields]] < STM
+
+    /** Inserts a new record into the table.
+      *
+      * @param record
+      *   The record to insert
+      * @return
+      *   The ID assigned to the new record, within the STM effect
+      */
+    def insert(record: Record[Fields])(using Frame): Id < STM
+
+    /** Updates an existing record in the table.
+      *
+      * @param id
+      *   The ID of the record to update
+      * @param record
+      *   The new record data
+      * @return
+      *   The previous record if it existed, None otherwise, within the STM effect
+      */
+    def update(id: Id, record: Record[Fields])(using Frame): Maybe[Record[Fields]] < STM
+
+    /** Updates an existing record or inserts a new one if it doesn't exist.
+      *
+      * @param id
+      *   The ID of the record to update or insert
+      * @param record
+      *   The record data
+      */
+    def upsert(id: Id, record: Record[Fields])(using Frame): Unit < STM
+
+    /** Removes a record from the table by its ID.
+      *
+      * @param id
+      *   The ID of the record to remove
+      * @return
+      *   The removed record if it existed, None otherwise, within the STM effect
+      */
+    def remove(id: Id)(using Frame): Maybe[Record[Fields]] < STM
+
+    /** Returns the current number of records in the table.
+      *
+      * @return
+      *   The number of records, within the STM effect
+      */
+    def size(using Frame): Int < STM
+
+    /** Checks if the table is empty.
+      *
+      * @return
+      *   true if the table contains no records, false otherwise, within the STM effect
+      */
+    def isEmpty(using Frame): Boolean < STM
+
+    /** Creates a snapshot of the entire table contents.
+      *
+      * @return
+      *   A map of all record IDs to their corresponding records, within the STM effect
+      */
+    def snapshot(using Frame): Map[Id, Record[Fields]] < STM
+
+end TTable
+
+object TTable:
+
+    /** Initializes a new basic table without indexing.
+      *
+      * @tparam Fields
+      *   The record structure for the table
+      * @return
+      *   A new TTable instance within the IO effect
+      */
+    def init[Fields: AsFields](using Frame): TTable[Fields] < IO =
```

Comment:
The distinction between `init` and `initNow` methods don't make much sense anymore after https://github.com/getkyo/kyo/pull/949. I'm adding only `init` here and will follow up to do the same for the other impls in the module